packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/InstanceManager.java

```diff
@@ -32,6 +36,8 @@ public class InstanceManager {
   // 0 <= n < 2^16.
   private static final long MIN_HOST_CREATED_IDENTIFIER = 65536;
   private static final long CLEAR_FINALIZED_WEAK_REFERENCES_INTERVAL = 30000;
+  private static final String TAG = "InstanceManager";
+  private static final String CLOSED_WARNING = "Method was called while the manager was closed.";
 
   /** Interface for listening when a weak reference of an instance is removed from the manager. */
   public interface FinalizationListener {
```

Comment:
This class looks like it has to check if the manager is closed in every method. That design pattern of requiring work in every method can be dangerous/tedious. It appears that each method has bespoke error conditions so that makes building a helper more difficult but here are some suggestions I would like you to consider. 

 1. Class method structure: 
    For every public method make a private method of the same name. The only logic in public methods is the check for close and error return value. 
As an example
```
public boolean containsInstance(Object instance) {
    if (isClosed()) {
      Log.w(TAG, CLOSED_WARNING);
      return false;
    }
    return constainsInstanceUnsafe(instance);
  }
// Probably a bad name. 
private boolean constainsInstanceUnsafe(Object instance) {
 return identifiers.containsKey(instance)
}

```

2.  Change the method definitions to only return the expected values and throw a typed error when the instance is called. This requires callers to catch the error but can make this class easier to maintain. This may or may not be combined with the first suggestion. What is nice this this behavior can be contained in a method and documentation added that says that every class in this method must call `verifyNotClosed()` first. 
3. Something else

Reply:
@bparrishMines what do you think about these alternatives?

Reply:
@reidbaker Looking to land this so I can make progress elsewhere. Can you file an issue for this or discuss it here, but approve this PR for now? I can make any changes in a follow up PR.

Reply:
"I don't see how this is much different than the current structure. It just moves the method logic to a private class. Is it not dangerous/tedious for the same reason?" 
Yes it is but at least the structure of the code makes it obvious that it is a design pattern. I am not sure the same is obvious with the current reading of the code. 

"To verify, this is suggesting throwing a checked exception when closed? This could be a good alternative solution, but it would require adding a try/catch block in a lot of places. " yes I tried to call that out in "This requires callers to catch the error but can make this class easier to maintain." 

The difficulties in this class are the result of InstanceManager exporting the need to check if it is closed first. I am not sure if that means it is either an acceptable burden to continue the cost up the tree or if it means that InstanceManager should be refactored to remove the need to check if closed before any interaction. 

@camsim99 can you find 30 minutes for you me and @bparrishMines to talk about this. I think the current structure is risky to maintain.  I will approve the pr as is since most of this update is unrelated.